spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/AtlasEntityWithDependencies.scala

```diff
@@ -17,13 +17,41 @@
 
 package com.hortonworks.spark.atlas
 
-import org.apache.atlas.model.instance.AtlasEntity
+import org.apache.atlas.model.instance.{AtlasEntity, AtlasObjectId}
 
-class AtlasEntityWithDependencies(
-    val entity: AtlasEntity,
-    val dependencies: Seq[AtlasEntityWithDependencies]) {
+trait AtlasReferenceable {
```

Comment:
I like adding `SAC` as classifying with Atlas one, but `Entity` is already used as object itself and this interface only promise object reference at most. Maybe `SACAtlasReferenceable` then?

Reply:
Maybe adding `SAC` as prefix would help on classifying whether the class/object is from SAC or Atlas.

Reply:
Adding SAC prefix surely helps.
I was confused with `AtlasEntityWithDependencies` also being a `AtlasReferenceable` (which I assumed to be a trait for something that can be referred) 
If the relation is like SAC entity can either be a reference or a real entity, naming the trait `SACAtlasEntity` helps. Since`SACAtlasEntityWithDependencies` holds a real entity, marking it `SACAtlasReferenceable` looked ambiguous to me.

Reply:
IMHO naming trait as `SACAtlasEntity` seems to give the feeling that it is directly linked with `AtlasEntity`.

So we're abstracting both `Entity` and `ObjectId` (which I just named as `EntityReference` as it is actually playing as) which I can't imagine good name to cover both. While working with such trait, I found the trait should at least provide `reference of entity` to let call site use it properly. That's why I picked `Referenceable` but as in many cases I'm not sure about better name, better name would be always welcome.